lib/tasks/server/livereload-server.js

```diff
@@ -76,21 +100,39 @@ module.exports = Task.extend({
   },
 
   writeErrorBanner: function(url) {
-    throw new SilentError('Livereload failed on ' + url + '.  It is either in use or you do not have permission.');
+    throw new SilentError('Livereload failed on ' + url +
+                          '.  It is either in use or you do not have permission.');
+  },
+
+  getDirectoryEntries: function(directory) {
+    return walkSync.entries(directory);
   },
 
   didChange: function(results) {
-    var filePath = path.relative(this.project.root, results.filePath || '');
+    var canTrigger = true;
+
+    if (this.project.liveReloadFilterPatterns.length > 0) {
+      var filePath = path.relative(this.project.root, results.filePath || '');
```

Comment:
I believe `canTrigger` may be flawed, as the filePath isn't going to always be right. It is very easy for many files to change, and get batched together...